Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterManager.java

```diff
@@ -118,7 +126,16 @@ public void runSpecificParameter(SubscriptionDetail parameter) throws IOExceptio
                     List<ClusterRawInfo> clusterRawInfoList = clusterOperation.listCluster(parameter);
                     if (clusterRawInfoList != null) {
                         for (ClusterRawInfo item : clusterRawInfoList) {
-                            IClusterDetail tempClusterDetail = new ClusterDetail(parameter, item);
+                            IClusterOperation clusterOperationForCluster = null;
+                            ClusterOperationNewAPIImpl probeClusterOperation = new ClusterOperationNewAPIImpl();
+                            if (isHDInsightNewSDKEnabled()
+                                    && probeClusterOperation.isProbeGetConfigurationSucceed(parameter, item.getId())) {
+                                clusterOperationForCluster = probeClusterOperation;
```

Comment:
Just `new ClusterDetail` here might be more clear, such as:
```suggestion
                                tempClusterDetail = new ClusterDetail(parameter, item, probeClusterNewApiOperation);
```